installer/conf/omsagent.d/mysql_logs.conf

```diff
@@ -0,0 +1,56 @@
+# MySql General Log
+
+<source>
+  type tail
+  tag oms.api.Mysql.General.logs
+  format /^((?<InitTime>[0-9]{0,6}\s{0,4}[0-9]{0,2}:[0-9]{0,2}:[0-9]{0,2})|\s{0,10})\s{0,4}+(?<Id>[0-9]{0,2})\s+(?<Command>\w+)\s+(?<Arguments>\z|.+)$/
+  path /var/log/mysql/mysql.log
+  pos_file /var/opt/microsoft/omsagent/state/var_log_mysql_log.pos
+</source>
+
+# MySql Error Log
+
+<source>
+  type tail
+  tag oms.api.Mysql.Error.logs
+  format /^(?<Timestamp>\d*\s{0,4}[0-9]{0,2}:[0-9]{0,2}:[0-9]{0,2})\s{0,4}(?<Command>[A-Za-z\:\[\]]{0,10})\s(?<Arguments>.*)$/
+  path /var/log/mysql/error.log
+  pos_file /var/opt/microsoft/omsagent/state/var_log_mysql_error_log.pos
+</source>
+
+# MySql Slow Query Log
+
+<source>
+  type tail
+  tag oms.api.Mysql.SlowQuery.logs
+  format multiline
+  format_firstline /^\#\s{0,4}User\@Host/
+  format1 /^\s{0,2}\#\s{0,2}User@Host:+(?<UserHost>.+)\n/
+  format2 /^\#\s{0,4}Query_time\:\s{0,4}(?<QueryTime>[\-\+]?[0-9]*(\.[0-9]+))\s{0,4}Lock_time\:\s{0,4}(?<LocalTime>[\-\+]?[0-9]*(\.[0-9]+))\s{0,4}Rows_sent\:\s{0,4}(?<RowsSent>\d*)\s{0,4}Rows_examined\:\s{0,4}(?<RowsExamined>\d*)\n/
+  format3 /\s{0,4}SET\s{0,4}timestamp\=(?<Timestamp>\d*)\;\s{0,4}\n/
+  format4 /(?<Query>^.*$)/
+  multiline_flush_interval 4s
+  path /var/log/mysql/mysql-slow.log
+  pos_file /var/opt/microsoft/omsagent/state/var_log_mysql_slow_log.pos
+</source>
+
+# Filter Plugin
+
+<filter oms.api.Mysql.**>
+  type filter_mysql_logs
+</filter>
+
+# Output Plugin
+
+<match oms.api.Mysql.**>
+  type out_oms_api
+  log_level info
+  buffer_chunk_limit 5m
+  buffer_type file
+  buffer_path /var/opt/microsoft/omsagent/state/out_oms_api_mysql_logs*.buffer
+  buffer_queue_limit 10
+  flush_interval 20s
+  retry_limit 10
+  retry_wait 5s
+  max_retry_wait 5m
```

Comment:
tip - you can define time_generated_field in the out_oms_api to specify the name of the time_generated field, so that the TimeGenerated field of the data will use the time from data instead of data ingestion time.